src/lib.jl

```diff
@@ -0,0 +1,5774 @@
+    end
+
+    function Lines_Get_X1()
+        ccall((:Lines_Get_X1, LIBRARY), Cdouble, ())
+    end
+
+    function Lines_New(Name)
+        ccall((:Lines_New, LIBRARY), Int32, (Cstring,), Name)
+    end
+
+    function Lines_Set_Bus1(Value)
+        ccall((:Lines_Set_Bus1, LIBRARY), Cvoid, (Cstring,), Value)
+    end
+
+    function Lines_Set_Bus2(Value)
+        ccall((:Lines_Set_Bus2, LIBRARY), Cvoid, (Cstring,), Value)
+    end
+
+    function Lines_Set_Length(Value)
+        ccall((:Lines_Set_Length, LIBRARY), Cvoid, (Cdouble,), Value)
+    end
+
+    function Lines_Set_LineCode(Value)
+        ccall((:Lines_Set_LineCode, LIBRARY), Cvoid, (Cstring,), Value)
+    end
+
+    function Lines_Set_Name(Value)
+        ccall((:Lines_Set_Name, LIBRARY), Cvoid, (Cstring,), Value)
+    end
+
+    function Lines_Set_Phases(Value)
+        ccall((:Lines_Set_Phases, LIBRARY), Cvoid, (Int32,), Value)
+    end
+
+    function Lines_Set_R1(Value)
+        ccall((:Lines_Set_R1, LIBRARY), Cvoid, (Cdouble,), Value)
+    end
+
+    function Lines_Set_X1(Value)
+        ccall((:Lines_Set_X1, LIBRARY), Cvoid, (Cdouble,), Value)
+    end
+
+    function Lines_Get_C0()
+        ccall((:Lines_Get_C0, LIBRARY), Cdouble, ())
+    end
+
+    function Lines_Get_C1()
+        ccall((:Lines_Get_C1, LIBRARY), Cdouble, ())
+    end
+
+    function Lines_Get_Cmatrix(ResultPtr, ResultCount)
+        ccall((:Lines_Get_Cmatrix, LIBRARY), Cvoid, (Ptr{Ptr{Cdouble}}, Ptr{Int32}), ResultPtr, ResultCount)
+    end
+
+    function Lines_Get_Cmatrix_GR()
+        ccall((:Lines_Get_Cmatrix_GR, LIBRARY), Cvoid, ())
+    end
+
+    function Lines_Get_R0()
+        ccall((:Lines_Get_R0, LIBRARY), Cdouble, ())
+    end
+
+    function Lines_Get_Rmatrix(ResultPtr, ResultCount)
+        ccall((:Lines_Get_Rmatrix, LIBRARY), Cvoid, (Ptr{Ptr{Cdouble}}, Ptr{Int32}), ResultPtr, ResultCount)
+    end
+
+    function Lines_Get_Rmatrix_GR()
+        ccall((:Lines_Get_Rmatrix_GR, LIBRARY), Cvoid, ())
+    end
+
+    function Lines_Get_X0()
+        ccall((:Lines_Get_X0, LIBRARY), Cdouble, ())
+    end
+
+    function Lines_Get_Xmatrix(ResultPtr, ResultCount)
+        ccall((:Lines_Get_Xmatrix, LIBRARY), Cvoid, (Ptr{Ptr{Cdouble}}, Ptr{Int32}), ResultPtr, ResultCount)
+    end
+
+    function Lines_Get_Xmatrix_GR()
+        ccall((:Lines_Get_Xmatrix_GR, LIBRARY), Cvoid, ())
+    end
+
+    function Lines_Set_C0(Value)
+        ccall((:Lines_Set_C0, LIBRARY), Cvoid, (Cdouble,), Value)
+    end
+
+    function Lines_Set_C1(Value)
+        ccall((:Lines_Set_C1, LIBRARY), Cvoid, (Cdouble,), Value)
+    end
+
+    function Lines_Set_Cmatrix(ValuePtr, ValueCount)
+        ccall((:Lines_Set_Cmatrix, LIBRARY), Cvoid, (Ptr{Cdouble}, Int32), ValuePtr, ValueCount)
+    end
+
+    function Lines_Set_R0(Value)
+        ccall((:Lines_Set_R0, LIBRARY), Cvoid, (Cdouble,), Value)
+    end
+
+    function Lines_Set_Rmatrix(ValuePtr, ValueCount)
+        ccall((:Lin
```

Comment:
Why is there a double indirection here? Most C API's that take arrays just need `Ptr{Cdouble}`.

Reply:
Ah. You asked about `YMatrix_SolveSystem`. [The header file](https://github.com/PMeira/dss_capi/blob/29b78fcb119900d743bc8e317f3070793ac821e7/include/v7/dss_capi.h#L5210) takes two levels of indirection because [the function in pascal](https://github.com/PMeira/dss_capi/blob/29b78fcb119900d743bc8e317f3070793ac821e7/src/v7/CAPI_YMatrix.pas#L131) uses `var` in one of the input arguments.

Reply:
@PMeira probably has more input here.

Reply:
`YMatrix_SolveSystem` is also like that (var to pointer) [in the DDLL](https://sourceforge.net/p/electricdss/code/HEAD/tree/trunk/Version8/Source/DDLL/DYMatrix.pas#l146):

```pascal
function SolveSystem(var NodeV:pNodeVarray): integer; cdecl;
begin
  Result:=ActiveCircuit[ActiveActor].Solution.SolveSystem(NodeV,ActiveActor);
end;
```

IIRC, I only changed `GetCompressedYMatrix` to allow it to return the raw matrix, without factorization. A lot of people use the dense SystemY equivalent and get surprised that it eats all their memory for large systems.